packages/atomic/src/components/search-box-suggestions/suggestions-common.ts

```diff
@@ -0,0 +1,41 @@
+import {SearchBox} from '@coveo/headless';
+import {VNode} from '@stencil/core';
+import {buildCustomEvent} from '../../utils/event-utils';
+import {Bindings} from '../../utils/initialization-utils';
+
+export interface SearchBoxSuggestionElement {
```

Comment:
When we want to make the "Custom Atomic component" API official, it means we will need to make the mechanism to extend "search box suggestions" provider public. And that can be a pretty tricky thing to get right.

A couple notes/thoughts:

* Since the `content` (the VNode) is ultimately rendered inside an `ul` /` li` in atomic-search-box, I guess that not all HTML elements would be valid ? I'm perfectly fine with this, by the way, but if there's any restriction, they need to be clear and (potentially) automatically handled (read: rejected) by the system so that users don't get random errors. Another approach would be to accept anything here, but to make it possible to control the "container" in `atomic-search-box` (being able to change `ul` / `li` to something else. This does add complexity, however.

* If someone wants to provide very rich/complex suggestion (eg: A product image with lots of metadata section), how would the keyboard navigation plays into that ? If you could navigate to different subsection on a suggestion. Should the system forbid that ?

* What happens if you want to have multiple clickable/interactive element in a given suggestion element ? Would the `onClick` cause a problem then ?

* Is the integration of that  `dispatchSearchBoxSuggestionsEvent` inside the `componentWillLoad` function as clean as it could be ? Could it be a more straightforward interface that you need to implement (or clever usage of an annotation).

Reply:
1. I want that limitation, the search box manages all the interactivity with the search box. The li is needed to be correctly accessible too.
2. Yes, a single element means a single possible interaction, more than that get's too complex for all our use cases.
3. Same as #2
4. What would you propose like interface? I will certainly be extended upon in the future, modify with our needs. It's not public yet and before it is we will have improved upon it to support instant results and recently viewed.